ray-curator/tests/core/test_get_ray_client.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+import time
+
+import pytest
+
+from ray_curator.core.client import RayClient
+
+# -----------------------------------------------------------------------------
+# Tests
+# -----------------------------------------------------------------------------
+
+
+# This test should be allowed to fail since it's not deterministic.
+@pytest.mark.xfail(strict=False, reason="Non-deterministic test due to Ray startup timing")
+def test_get_ray_client_single_start():
+    with tempfile.TemporaryDirectory(prefix="ray_test_single_") as ray_tmp:
+        # Clear the environment variable RAY_ADDRESS
+        os.environ.pop("RAY_ADDRESS", None)
+        client = RayClient(ray_temp_dir=ray_tmp)
+        client.start()
+        time.sleep(10)  # Wait for ray to start.
+        with open(os.path.join(ray_tmp, "ray_current_cluster")) as f:
+            assert f.read() == f"127.0.0.1:{client.ray_port}"
+        client.stop()
+
```

Comment:
Can you add a asset here to verify stop actually worked ?

Reply:
Hmm there is no easy way to verify that the stop actually worked.